src/test/kotlin/uk/gov/dluhc/registercheckerapi/database/repository/RegisterCheckRepositoryTest.kt

```diff
@@ -343,25 +335,13 @@ internal class RegisterCheckRepositoryTest : IntegrationTest() {
         @Test
         fun `should exclude register checks with statuses other than pending`() {
             // Given
-            val otherStatuses = listOf(
-                CheckStatus.ARCHIVED,
-                CheckStatus.NO_MATCH,
-                CheckStatus.EXACT_MATCH,
-                CheckStatus.PARTIAL_MATCH,
-                CheckStatus.MULTIPLE_MATCH,
-                CheckStatus.TOO_MANY_MATCHES,
-                CheckStatus.PENDING_DETERMINATION,
-                CheckStatus.EXPIRED,
-                CheckStatus.NOT_STARTED,
-            )
-            registerCheckRepository.saveAll(
-                otherStatuses.map { status ->
-                    buildRegisterCheck(status = status)
-                }
-            ).run {
-                setDateCreatedToOneHundredSecondsAgo()
-                registerCheckRepository.saveAll(this)
-            }
+            CheckStatus.entries
+                .filterNot { it == CheckStatus.PENDING }
+                .map { status -> buildRegisterCheck(status = status) }
+                .let(registerCheckRepository::saveAll)
+
+            // Sleep to guarantee date_created in past
+            Thread.sleep(1000)
```

Comment:
Aside from the issue with the job integration test, it'd be nice if we could mock the time used for these sorts of tests to avoid these sleep statements. Maybe one to think about for optimising services?

Reply:
Something else I didn't experiment with is using the `@DataJpaTest` annotation instead of extending `IntegrationTest` for our repository tests. I think it should be much lighter-weight

Reply:
Yeah would rather keep against a real DB as in-memory ones are only an approximation. For OS we could consider writing our own equivalent implementations allowing clock customisation